### PR TITLE
[TiDB] Add a new mode `prefer-leader` for `replica_read_mode`.

### PR DESCRIPTION
**Pull request by LykxSassinator**
Signed-off-by: Lucasliang <nkcs_lykx@hotmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add a new mode `prefer-leader` for the system variable `replica_read_mode`.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12865
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: follower-read.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ This variable is used to set the expected data read mode.
 - When the value of `tidb_replica_read` is set to `leader` or an empty string, TiDB maintains its default behavior and sends all read operations to the leader replica to perform.
 - When the value of `tidb_replica_read` is set to `follower`, TiDB selects a follower replica of the Region to perform all read operations.
 - When the value of `tidb_replica_read` is set to `leader-and-follower`, TiDB can select any replicas to perform read operations. In this mode, read requests are load balanced between the leader and follower.
+- When the value of `tidb_replica_read` is set to `prefer-leader`, TiDB selects the leader replica as priority to perform read operations. And if the leader replica is obviously slow on processing such as by Disk / Net performance jitter, TiDB will select other available follower replicas to perform.
```
Comment (TomShawn) [inline follower-read.md:38]:
```suggestion
- When the value of `tidb_replica_read` is set to `prefer-leader`, TiDB prefers to select the leader replica to perform read operations. If the leader replica is obviously slow in processing read operations (such as caused by disk or network performance jitter), TiDB will select other available follower replicas to perform read operations.
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Comment by LykxSassinator** [comment]
@TomShawn As all related PRs merged, this pr can be merged now.

**Comment by TomShawn** [comment]
> Details

This PR needs approval by @easonn7.

**Comment by LykxSassinator** [comment]
> > Details
> 
> This PR needs approval by @easonn7.

cc. The related PR: https://github.com/pingcap/docs-cn/pull/12865 has been approved.

File: develop/dev-guide-use-follower-read.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ If your TiDB cluster is deployed across districts or data centers, different rep
 <SimpleTab groupId="language">
 <div label="SQL" value="sql">
 
-To enable Follower Read, set the variable `tidb_replica_read` (default value is `leader`) to `follower`, `leader-and-follower`, `closest-replicas`, or `closest-adaptive`:
+To enable Follower Read, set the variable `tidb_replica_read` (default value is `leader`) to `follower`, `leader-and-follower`, `prefer-leader`, closest-replicas`, or `closest-adaptive`:
```
Comment (TomShawn) [inline develop/dev-guide-use-follower-read.md:47]:
```suggestion
To enable Follower Read, set the variable `tidb_replica_read` (default value is `leader`) to `follower`, `leader-and-follower`, `prefer-leader`, `closest-replicas`, or `closest-adaptive`:
```

**Pull request merged**